inst/@symfun/mtimes.m

```diff
@@ -58,7 +58,8 @@
 
 function h = mtimes(f, g)
   [vars, s1, s2] = helper_symfun_binops(f, g);
-  h = symfun(s1 * s2, vars);
+  res = s1 * s2;
+  h = symfun(res, symvar(res));
```

Comment:
Well, this is most confusing part for me... When we are doing any arithmetic operator. The result will definitely have variables from both the symbols involved in that operation. In `helper_symfun_binops`, it's not allowed to do operations on 2 symfuns if they dont' have exactly same variables!! Also, the variables of the `symfun` are only considered as final variables in any operation!! This definitely looks wrong to me. I was tempted to change `helper_symfun_binops` to take a union of the sets of variables in each of the operands involved but I currently did this only for `mtimes`.  
Am I missing something? Why isn't operations for symfuns with different variables allowed? Why are variables only from the symfun operand considered as variables in final expression?